Better runtime errors

If my test throws a runtime error, it gets printed as an `AssertionError`, which swallows the stack trace. For example,
```js
var Promise = require("bluebird");
test("foo", Promise.coroutine(function*(t){
    throw new Error("asd");
}));
```
only prints
```
1. foo
  AssertionError: Promise rejected → Error: asd
```
It would be really helpful to get a stack trace here.

--- a/api.js
+++ b/api.js
@@ -84,8 +84,14 @@ Api.prototype._handleTest = function (test) {
 			if (test.error.originalMessage) {
 				message = test.error.originalMessage + ' ' + message;
 			}
+
 			test.error.message = message;
 		}
+
+		if (test.error.name !== 'AssertionError') {
+			test.error.message = 'failed with "' + test.error.message + '"';
+		}
+
 		this.errors.push(test);
 	} else {
 		test.error = null;

--- a/lib/test.js
+++ b/lib/test.js
@@ -118,12 +118,7 @@ Test.prototype.run = function () {
 				self.exit();
 			})
 			.catch(function (err) {
-				self._setAssertError(new assert.AssertionError({
-					actual: err,
-					message: 'Promise rejected → ' + err,
-					operator: 'promise'
-				}));
-
+				self._setAssertError(err);
 				self.exit();
 			});
 	} else if (!this.metadata.callback) {

--- a/test/promise.js
+++ b/test/promise.js
@@ -245,7 +245,8 @@ test('reject', function (t) {
 		});
 	}).run().catch(function (err) {
 		t.ok(err);
-		t.is(err.name, 'AssertionError');
+		t.is(err.name, 'Error');
+		t.is(err.message, 'unicorn');
 		t.end();
 	});
 });

